sdks/python/apache_beam/utils/dependency.py

```diff
@@ -435,7 +435,13 @@ def _stage_dataflow_sdk_tarball(sdk_remote_location, staged_path, temp_dir):
     _dependency_file_copy(sdk_remote_location, staged_path)
   elif sdk_remote_location == 'pypi':
     logging.info('Staging the SDK tarball from PyPI to %s', staged_path)
-    _dependency_file_copy(_download_pypi_sdk_package(temp_dir), staged_path)
+    import pkg_resources as pkg
+    try:
+      _dependency_file_copy(_download_pypi_sdk_package(temp_dir), staged_path)
+    except pkg.DistributionNotFound:
+      raise RuntimeError('Unable to stage SDK tarball. '
```

Comment:
I am not sure what would be a clear error message. There is no need to mention remote location that is an SDK internal.

Perhaps : 'Please set --sdk_location or install a valid GOOGLE_PACKAGE_NAME distribution.'